build/dnsmasq/local.mog

```diff
@@ -0,0 +1,20 @@
+#
+# This file and its contents are supplied under the terms of the
+# Common Development and Distribution License ("CDDL"), version 1.0.
+# You may only use this file in accordance with the terms of version
+# 1.0 of the CDDL.
+#
+# A full copy of the text of the CDDL should have accompanied this
+# source. A copy of the CDDL is also available via the Internet at
+# http://www.illumos.org/license/CDDL.
+#
+# Copyright 2022 OmniOS Community Edition (OmniOSce) Association.
+
+dir group=$(GROUP) mode=0770 owner=$(USER) path=var/$(BASEDIR)
+
+group groupname=$(GROUP) gid=99
+user ftpuser=false username=$(USER) uid=99 group=$(GROUP) \
+    gcos-field="dnsmasq - DNS Forwarder" \
+    home-dir=/var/$(BASEDIR) password=NP
+
```

Comment:
To stop the configuration file being overwritten on package updates, you will need to set a preserve flag on it. There are various options (See https://man.omnios.org/pkg.7 ) but I usually favour `renamenew` - so something like:

```
<transform file path=etc/ -> set preserve renamenew>
```

You should also configure the service to automatically restart if the binary changes, which is:

```
<transform file path=$(PREFIX)/sbin -> \
    set restart_fmri svc:/network/dns/$(PROG):default >
```